src/Slist.hs

```diff
@@ -1892,3 +1900,74 @@ insert = insertBy compare
 insertBy :: (a -> a -> Ordering) -> a -> Slist a -> Slist a
 insertBy f a Slist{..} = Slist (L.insertBy f a sList) (sSize + 1)
 {-# INLINE insertBy #-}
+
+----------------------------------------------------------------------------
+-- Generic fuctions
+----------------------------------------------------------------------------
+
+-- | O(1)
+-- The 'genericLength' function is an overloaded version of 'length'.  In
+-- particular, instead of returning an 'Int', it returns any type which is
+-- an instance of 'Num'.
+genericLength :: Num i => Slist a -> i
+genericLength = fromIntegral . length
+{-# INLINE genericLength #-}
+
+-- | @O(i) | i < n@ and @O(1) | otherwise@.
+-- The 'genericTake' function is an overloaded version of 'take', which
+-- accepts any 'Integral' value as the number of elements to take.
+genericTake :: Integral i => i -> Slist a -> Slist a
+genericTake n Slist{..}
```

Comment:
Sorry for being picky, but can we also name it `i` everywhere for consistency :smile:

Reply:
I see that `genericUnsafeAt` is using `i` as well.

In my head, `n` usually for length or the last index in the list, but `i` can be somewhere in between. Fell free to update if you noticed the inconsistent in the existing code as well. But, in the end, it's not important at all. This PR is not approved mostly because it lacks testing functions... So, don't worry too much about this :slightly_smiling_face: